libbeat/docs/command-reference.asciidoc

```diff
@@ -0,0 +1,583 @@
+
+endif::[]
+
+<<run-command,`run`>>::
+{run-command-short-desc}.
+
+<<setup-command,`setup`>>::
+{setup-command-short-desc}.
+
+<<test-command,`test`>>::
+{test-command-short-desc}.
+
+<<version-command,`version`>>::
+{version-command-short-desc}.
+
+Also see <<global-flags,Global flags>>.
+
+[[export-command]]
+==== `export` command
+
+{export-command-short-desc}. You can use this
+command to quickly view your configuration or the contents of the index
+template.
+
+*SYNOPSIS*
+
+["source","sh",subs="attributes"]
+----
+{beatname_lc} export SUBCOMMAND [FLAGS]
+----
+
+
+*SUBCOMMANDS*
+
+*`config`*::
+Exports the current configuration to stdout. If you use the `-c` flag, this
+command exports the configuration that's defined in the specified file.
+
+[[template-subcommand]]
+*`template`*::
+Exports the index template to stdout. You can specify the `--es.version` and
+`--index` flags to futher define what gets exported.
+
+*FLAGS*
+
+*`--es.version VERSION`*::
+When specified along with <<template-subcommand,`template`>>, exports an index
+template that is compatible with the specified version.
+
+
+//REVIEWERS: Is the above statement true? The command-line help was too vague (said "Elastic version"). When I run export template with --es.version specified, nothing seems to happen.
+
+*`-h, --help`*::
+Shows help for the `export` command.
+
+*`--index BASE_NAME`*::
+When specified along with <<template-subcommand,`template`>>, sets the base name
+to use for the index template. If this flag is not specified, the default base
+name is +{beatname_lc}+.
+
+{global-flags}
+
+*EXAMPLES*
+
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} export config
+{beatname_lc} export template --es.version {stack-version} --index myindexname
+-----
+
+
+[[help-command]]
+==== `help` command
+
+{help-command-short-desc}. If no command is specified, shows help for the
+`run` command.
+
+*SYNOPSIS*
+
+["source","sh",subs="attributes"]
+----
+{beatname_lc} help COMMAND_NAME [FLAGS]
+----
+
+
+*`COMMAND_NAME`*::
+Specifies the name of the command to show help for.
+
+*FLAGS*
+
+*`-h, --help`*:: Shows help for the `help` command.
+
+{global-flags}
+
+*EXAMPLE*
+
+["source","sh",subs="attributes"]
+-----
+{beatname_lc} help export
+-----
+
+ifeval::[("{beatname_lc}"=="filebeat") or ("{beatname_lc}"=="metricbeat")]
+
+[[modules-command]]
+==== `modules` command
+
+{modules-command-short-desc}. You can use this command to enable and disable
+specific modules. The changes you make with this command are persisted and
+used for subsequent runs of {beatname_uc}. To see which modules are enabled
+and disabled, run the `list` subcommand.
+
+*SYNOPSIS*
+
+["source","sh",subs="attributes"]
+----
+{beatname_lc} modules SUBCOMMAND [FLAGS]
+----
+
+
+*SUBCOMMANDS*
+
+*`disable MODULE_LIST`*::
+Disables the modules specified in the space-separated list.
+
+//REVIEWERS: It seems odd to me that we usually expect comma-separated lists, but here we expect a space-separated list.
```

Comment:
Ah OK. I didn't understand that distinction (and couldn't really take the extra time to test all permutations).